### DISCUSSION TITLE
Scripting elisp: wait for forge-pull to finish

### DISCUSSION THREAD

**Discussion by bascht** [magit/forge#597] [Support] [answered]
Hey everyone. I tried to create a small wrapper function for myself, to create GitLab merge requests on our private instance. Everything works so far, but I'm failing to wrap my code around `(forge-pull)`.

It looks like a call to `(forge-pull)` will run in the background and it doesn't block. Other functions, e.g. in `magit` do have `-async` counterparts, but I didn't find anything for `forge` and there are no hooks for me to latch onto.

How would I implement something like:

```elisp
(create-mr)                ; which already works
(forge-pull)               ; fetch the newly created merge request
(do-something-with-the-mr) ; which will currently fail since the `forge-pull` is not done
```

**Comment by tarsius** [comment, marked as answer]
You can advise `forge--git-fetch` for this purpose, it is called after fetching api data. I might add a hook to that eventually, but am too busy right now, to make any design decisions. (I plan to do some refactoring in this area, soonish, and don't want to complicate that with any hasty decisions, I cannot back out anymore.)
